typhon/cloudmask/aster.py

```diff
@@ -617,79 +619,100 @@ def sensor_angles(self, sensor="vnir"):
+        # end of the image.
+        swath_width = np.mean(swath_widths[swath_widths > 4200])
+
+        n_angles = n * FOV[sensor] / swath_width + P[sensor]
+        azimuth = np.full_like(field, np.nan)
+
+        if channel is not "3B":
+            zenith = abs(n_angles)
+
+            if flyingdir is "DE":
+                #n_angles = 
+                azimuth[n_angles > 0] = 90 + S
+                azimuth[n_angles <= 0] = 270 + S
+            else:
+                azimuth[n_angles < 0] = 90 + S
+                azimuth[n_angles >= 0] = 270 + S
         else:
-            raise ValueError("ASTER sensors are named: vnir, swir, tir.")
-
-        # Instantaneous FOV (IVOF)
-        # IFOV = FOV / cloudmask.shape[1]
-
-        # Construct n-array indexing pixels n=- right and n=+ left from the image
-        # center and perpendicular to flight direction.
-        n = np.zeros(np.shape(field))
-        nswath = []
-        # assign indices n to pixel, neglegting the frist and last 5 swath rows
-        for i in np.arange(start=5, stop=field.shape[0] - 5, step=1, dtype=int):
-            # for all scan rows: extract first and last not-NaN index
-            ind1 = next(x[0] for x in enumerate(field[i]) if ~np.isnan(x[1]))
-            # print('Index 1: ', ind1)
-            ind2 = field.shape[1] - next(
-                x[0] for x in enumerate(field[i, 5:-5][::-1]) if ~np.isnan(x[1])
-            )
-            ind_mid = (ind1 + ind2) / 2
-            nswath.append(ind2 - ind1 + 1)
-            # create indexing arrays for right and left side of swath
-            right_arr = np.arange(start=-round(ind_mid), stop=0, step=1, dtype=int) * -1
-            left_arr = (
-                np.arange(
-                    start=1, stop=field.shape[1] - round(ind_mid) + 1, step=1, dtype=int
-                )
-                * -1
-            )
-            n[i] = np.asarray(list(right_arr) + list(left_arr))
-
-        # add the first and last 5 scan lines as dupliates of the 6th and 6th-last row.
-        n[:5] = n[5]  # first
-        n[-5:] = n[-6]  # last
-
-        n[np.isnan(field)] = np.nan
-
-        # instantaneous field of view (IFOV)
-        FOV_S = FOV / np.cos(np.deg2rad(S))  # correct for map orientation
-        IFOV = FOV_S / np.median(nswath)  # median scan line width
-
-        # Thresholding index for separating left and right from nadir in
-        # azimuth calculation.
-        n_az = n * IFOV + P
+            h = 705000 # in km above the equator
+            zenith = np.rad2deg(np.arctan(np.sqrt((h * np.tan(np.deg2rad(P[sensor]))
+                                                   + 15 * n)**2
+                                                  + (h * np.tan(np.deg2rad(27.6))
+                                                     / np.cos(np.deg2rad(P[sensor])))**2
+                                                 ) / h
+                                         )
+                               )
```

Comment:
Did you or your IDE/code formatter chose that style? ;)
You could add a new line before `h * np.tan...` this gives you way more space to operate in the next line

Reply:
personal choice :) but I'm open to the `back` suggestions